sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala

```diff
@@ -242,4 +243,16 @@ object DataSourceUtils {
       options
     }
   }
+
+  def getPartitionFiltersAndDataFilters(
+      partitionColumns: Seq[Attribute],
```

Comment:
It seems we can simplify the caller side if this method takes `partitionSchema: StructType`